app-frontend/src/app/pages/market/search/search.controller.js

```diff
@@ -1,126 +1,156 @@
+/* global _ */
+
 export default class MarketSearchController {
     constructor( // eslint-disable-line max-params
-        $log, $state
+        $log, $state, toolService, toolTagService, toolCategoryService
     ) {
         'ngInject';
+        this.toolService = toolService;
+        this.toolTagService = toolTagService;
+        this.toolCategoryService = toolCategoryService;
         this.$state = $state;
         this.$log = $log;
+        this.nonSpaceDelimiter = '|';
```

Comment:
See my comment below, I think this shouldn't be necessary if we're using UUIDs for the tags.

I don't remember the rationale, but I think because `category` is pretty static (users will never define them) might be why they have slugs, but because tags can be edited (possibly) might be why they were omitted.